src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -61,22 +61,16 @@ internal DefaultRazorLanguageServerCustomMessageTarget(TrackingLSPDocumentManage
             _documentManager = documentManager;
         }
 
-        public override async Task UpdateCSharpBufferAsync(JToken token, CancellationToken cancellationToken)
+        public override async Task UpdateCSharpBufferAsync(UpdateBufferRequest request, CancellationToken cancellationToken)
         {
-            if (token is null)
-            {
-                throw new ArgumentNullException(nameof(token));
-            }
-
             await _joinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
-            UpdateCSharpBuffer(token);
+            UpdateCSharpBuffer(request);
```

Comment:
Still need the null check for request so we can throw early before switching to the main thread.